### ISSUE TITLE
***.nupkg in ppy/osu releases has changed since 2024.906.1

### ISSUE THREAD

**Issue by Insulinocytus** [ppy/osu#29783]
### Type

Other

### Bug description

An error occurred when I used Scoop to install the game. It tried to decompress the folder with the path `lib/net45` from `osulazer-2024.906.2-full.nupkg` and reported an error.
This means that the folder structure in `osulazer-2024.906.2-full.nupkg` has changed.

So I downloaded the latest version of `osulazer-<version>-full.nupkg` from ppy/osu releases and compared them. I found that starting from `2024.906.1`, the content of nupkg has changed as follows. And there is no explanation for this in the Release Note.

![image](https://github.com/user-attachments/assets/120d7aa5-23e7-4b56-8589-e903f39e4ce9)
![image](https://github.com/user-attachments/assets/29e19c72-7b79-41ae-8e27-21b4d2a0e7f6)

This folder was renamed
`osulazer-2024.817.0-full.nupkg/lib/net45` => `osulazer-2024.906.1-full.nupkg/lib/app`

But from the source code
https://github.com/ppy/osu/blob/e88083adc17a383ae6e285d378988c01c40985ad/osu.Desktop/osu.nuspec#L19-L24
It seems `/lib/net45` is correct?

`osulazer-2024.817.0-full.nupkg/osulazer.nuspec`
```xml
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
  <metadata>
    <id>osulazer</id>
    <version>2024.817.0</version>
    <title>osu!</title>
    <authors>ppy Pty Ltd</authors>
    <owners>Dean Herbert</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <icon>icon.png</icon>
    <projectUrl>https://osu.ppy.sh/</projectUrl>
    <iconUrl>https://github.com/ppy/osu/blob/master/assets/lazer-nuget.png?raw=true</iconUrl>
    <description>A free-to-win rhythm game. Rhythm is just a *click* away!</description>
    <releaseNotes>&lt;![CDATA[
&lt;p&gt;testing&lt;/p&gt;

]]&gt;</releaseNotes>
    <copyright>Copyright (c) 2024 ppy Pty Ltd</copyright>
    <language>en-AU</language>
    <machineArchitecture>x64</machineArchitecture>
  </metadata>
</package>

```

`osulazer-2024.906.1-full.nupkg/osulazer.nuspec`
```xml
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
<metadata>
<id>osulazer</id>
<title>osu!(lazer)</title>
<description>osu!(lazer)</description>
<authors>osulazer</authors>
<version>2024.906.1</version>
<channel>win</channel>
<mainExe>osu!.exe</mainExe>
<os>win</os>
<rid>win-x64</rid>
<shortcutLocations>Desktop,StartMenuRoot</shortcutLocations>
<shortcutAmuid>velopack.osulazer</shortcutAmuid>
<machineArchitecture>x64</machineArchitecture>
</metadata>
</package>
```

Are these changes intentional?

### Version

2024.906.1

### Logs

Since this problem does not occur while the game is running, there is no log.

**Comment by bdach** [comment]
This is going to invariably be a side effect of switching from squirrel to velopack. Unless you can describe a _very_ good reason why this matters, I don't think we're going to be doing anything about it.

**Comment by bdach** [comment]
Also the game has an auto-update facility, so whatever this "scoop" thing is that you're using, probably don't?

**Comment by Insulinocytus** [comment]
> This is going to invariably be a side effect of switching from squirrel to velopack.

Got it. Thank you for your reply.

>  Unless you can describe a very good reason why this matters, I don't think we're going to be doing anything about it.
Also the game has an auto-update facility, so whatever this "scoop" thing is that you're using, probably don't?

I'm sorry I didn't express my meaning correctly.

Let me explain what happened.
- [Scoop](https://scoop.sh/) is a package manager for Windows, similar to Ubuntu `apt`, and is maintained by the community.
- When I used Scoop to update OSU, I found that the game could not be installed normally because the file path inside nupkg had changed (`/lib/net45` => `/lib/app`).
- I would like to submit a PR to Scoop to fix this issue
- I need to explain in Scoop's PR why I changed `/lib/net45` to `/lib/app`
- I searched for OSU's Release Note, but because I don't know much about C# desktop program development, I can't explain why `/lib/net45` became `/lib/app`
- I'm opening this issue to ask for help in determining this is a known change and not a BUG

Now that you tell me "This is going to invariably be a side effect of switching from squirrel to velopack.", I can go to Scoop and continue to submit PR. Thank you again for your reply.
